rfcs/0006-merkle-tree/0006-merkle-tree.md

```diff
@@ -57,16 +57,16 @@ Suppose a CBMT with *n* items, the size of the array would be *2n-1*, the index
 
 ## Merkle Proof
 
-Merkle Proof can provide a proof for existence of one or more items. Only sibling of the nodes along the path that form leaves to root, excluding the nodes already in the path, should be included in the proof. We also specify that ***the nodes in the proof is presented in descending order***(with this, algorithms of proof's generation and verification could be much simple). For example, if we want to show that `[T1, T4]` is in the list of 6 items above, only nodes `[T5, T0, B3]` should be included in the proof.
+Merkle Proof can provide a proof for existence of one or more items. Only sibling of the nodes along the path that form leaves to root, excluding the nodes already in the path, should be included in the proof. We also specify that ***the nodes in the proof is presented in descending order***(with this, algorithms of proof's generation and verification could be much simple). The indexes of items that need to prove is also included in the proof, and ***presented in ascending order by corresponding hash***. For example, if we want to show that `[T1, T4]` is in the list of 6 items above, only nodes `[T5, T0, B3]` and indexes `[9, 6]` should be included in the proof.
 
-### Proof Sturct
+### Proof Structure
 
 The schema of proof struct is:
 
 ```
 table Proof {
-  // size of items in the tree
-  size: uint32;
```

Comment:
The attacker can create a fake item which still passes the proof verification if we omit item size here. e.g.
A merkle tree with 4 items:
```
                B0
        ┌───────┴───────┐
        B1              B3
    ┌───┴───┐       ┌───┴───┐
    T0      T1      T2      T3
```

A fake item with brute forced hash can be created like this diagram:
```
                B0
        ┌───────┴───────┐
        B1              B3
    ┌───┴───┐       ┌───┴───┐
    T0      T1      T2      T3
                  ┌─┴─┐
                 T4 Brute-forced-hash
```